src/Program.cs

```diff
@@ -48,60 +62,35 @@ public static void Main(string[] args)
                 {
                     try
                     {
-
                         var upstreamResponse = await SendUpstreamRequest(context.Request);
+                        
+                        // Attempt to remove the response selection header and use its value to handle the response selection.
+                        if (context.Request.Headers.Remove(_responseSelectionHeader, out var selection))
+                        {
+                            string optionDescription = _selectionDescriptions.FirstOrDefault(kvp => kvp.Key.Equals(selection)).Value;
```

Comment:
If the selection header exists but the value is invalid, I would print a warning.